app/js/createCSV.ts

```diff
@@ -4,47 +4,29 @@
  * @param {info} Object - object of user data
  * @param {userName} string  - name of user in database
  * @param {userPercentage} number -user percentage score on taken test
+ * @param {score} number -user score on taken test
  */
-function createCSV(info: any, userName: string, userPercentage: number) {
-    
-    //declare regular expressions for characters to remove from the csv
-    let arrayOfReplaces = [/\\/g, /"/g, /{/g, /}/g,]
-    let questionsAndAnswers = info.data.answers
-    let questionNumbersAndAnswers = []
-    let csv = 'Name,Score,%'
-    
-    //iterate over the users answers and remove unwanted characters
-    arrayOfReplaces.forEach((expression) => {
-        questionsAndAnswers = questionsAndAnswers.replace(expression, '')
-    })
+function createCSV(info: Object, userName: string, userPercentage: number, score: number) {
 
-    //split the stripped file by the commas
-    questionsAndAnswers = questionsAndAnswers.split(',')
-
-    //split the stripped file by the seperator between question number and the answer
-    questionsAndAnswers.forEach((answer) => {
-        questionNumbersAndAnswers.push(answer.split(':'))
-    })
-    
+    let csv = "Name,Score,%"
 
     //add each question to the titles of the csv
-    questionNumbersAndAnswers.forEach((eachQuestion) => {
+   for(let eachQuestion in info) {
+       let questionString = info[eachQuestion].question
+       let replacedString = questionString.replace(/,/g,"")
+        csv += `, ${eachQuestion}: ${replacedString}`
+    }
 
-        csv += `, ${eachQuestion[0]}`
-        
-    })
-    
-    csv += '\n'
+    csv += "\n"
 
     //enter the username, score and percentage under the relevant heading
-    csv += `${userName},${info.data.score},${userPercentage}`
-
-    //add the question answer under the relevant heading in the csv
-    questionNumbersAndAnswers.forEach((eachQuestion) => {
+    csv += `${userName},${score},${userPercentage}`
 
-        csv += `, ${eachQuestion[1]}`
-
-    })
+   //add the question answer under the relevant heading in the csv
+    for(let eachQuestion in info) {
+        csv += `,User Response: ${info[eachQuestion].userAnswer} - ${info[eachQuestion].correct}`
```

Comment:
Remove the `User Response: ` bit so that the final string looks like this:
`1 - incorrect`